tests/experimental/wgit/test_signal_sparsity.py

```diff
@@ -6,50 +6,63 @@
 import pytest
 import torch
 
-from fairscale.experimental.wgit.signal_sparsity import SignalSparsity, _get_k_for_topk, _top_k_total_size
-
-TRIALS = 5
-DIM_LIST = [None, 0, 1, 2]
-SPARSITY_ATOL = 8e-2
-FFT_RECONS_ATOL = 2e-3
-
-
-@pytest.fixture
-def sst_and_default_sps():
-    """A fixture returning a function that calculates the sst from pytorch operations
-    and compares it with the sst from dense_to_sst method."""
-
-    def assert_and_get_sps_sst(tensor, percent, element, dim):
-        dense_freq = torch.fft.fft(tensor)
-
-        real_dense_freq = dense_freq.real
-        sparser_2d = SignalSparsity(
-            sst_top_k_percent=percent, sst_top_k_element=element, sst_top_k_dim=dim, dst_top_k_percent=100
-        )
-        top_k_total_size = _top_k_total_size(tensor, dim)
-        k = _get_k_for_topk(percent, element, top_k_total_size)
-        default_sst = torch.zeros_like(dense_freq)
-
-        orig_shape = dense_freq.shape
-        if dim is None and len(orig_shape) > 1:
-            default_sst = default_sst.reshape(-1)
-            real_dense_freq = real_dense_freq.reshape(-1)
-            dense_freq = dense_freq.reshape(-1)
-            dim = -1
-
-        _, i = real_dense_freq.abs().topk(k, dim=dim)
-        default_sst = default_sst.scatter(dim, i, dense_freq.gather(dim, i)).reshape(orig_shape)
-
-        # verify if sst only has top-k values across the right dim
-        dim = None if (dim == -1) else dim
-        assert all((default_sst.abs().count_nonzero(dim) == k).flatten())
-
-        sst = sparser_2d.dense_to_sst(tensor)
-        assert all((default_sst == sst).flatten())
-        assert sst.shape == tensor.shape
-        return default_sst, sst
-
-    return assert_and_get_sps_sst
+from fairscale.experimental.wgit.signal_sparsity import SignalSparsity
+
+
+def expected_io_vals():
+    """Helper function to create and return a list of tuples of the form:
+    (in_tensor, expected_tensor, dim, percent) to be used as parameters for tests. The top-k
+    value for the expected SST was set at 2.
```

Comment:
not sure where is the hardcoded "2" this comment is talking about. Did I misunderstand it?

Reply:
Oh right. I was trying to express that these expected output tensors were created when we select top_k_element = 2 and dim=dim from the input tensor. Basically, these outputs are only expected when top_k_element = 2 (or equivalent top_k_percents) is set on Signal Sparsity. Hence, the 2. Do you suggest a better way to express this? May be: `The top-k value for the expected SST "should be" set at 2.` ?


```

Reply:
Why not just omit this info here but return 2 as the number of elements in top-k at the end of this function as a test_param? That way, there is no hardcoding and 2 is close to the expected value here in the same function at least. thoughts?

Reply:
Sounds much better tbh. Making the change. Thanks!